test/internal/bookmark.test.js

```diff
@@ -34,35 +34,60 @@ describe('#unit Bookmark', () => {
     ])
 
     expect(bookmark.isEmpty()).toBeFalsy()
-    expect(bookmark.maxBookmarkAsString()).toEqual('neo4j:bookmark:v1:tx3')
+    expect(bookmark.values()).toEqual([
+      'neo4j:bookmark:v1:tx1',
+      'neo4j:bookmark:v1:tx2',
+      'neo4j:bookmark:v1:tx3'
+    ])
   })
 
   it('should be possible to construct bookmark from null', () => {
     const bookmark = new Bookmark(null)
 
     expect(bookmark.isEmpty()).toBeTruthy()
-    expect(bookmark.maxBookmarkAsString()).toBeNull()
+    expect(bookmark.values()).toEqual([])
   })
 
   it('should be possible to construct bookmark from undefined', () => {
     const bookmark = new Bookmark(undefined)
 
     expect(bookmark.isEmpty()).toBeTruthy()
-    expect(bookmark.maxBookmarkAsString()).toBeNull()
+    expect(bookmark.values()).toEqual([])
   })
 
   it('should be possible to construct bookmark from an empty string', () => {
     const bookmark = new Bookmark('')
 
     expect(bookmark.isEmpty()).toBeTruthy()
-    expect(bookmark.maxBookmarkAsString()).toBeNull()
+    expect(bookmark.values()).toEqual([])
   })
 
   it('should be possible to construct bookmark from empty array', () => {
     const bookmark = new Bookmark([])
 
     expect(bookmark.isEmpty()).toBeTruthy()
-    expect(bookmark.maxBookmarkAsString()).toBeNull()
+    expect(bookmark.values()).toEqual([])
+  })
+
+  it('should be possible to construct bookmark from nested arrays', () => {
```

Comment:
Is it legal to have `session.lastBookmark` to return `null` or `undefined`?
If so, maybe also add a test for testing `null` and/or `undefined` inside bookmark arrays?
Basically to simulate how a user would give bookmark arrays to session.

Reply:
There wouldn't be any `null` or `undefined` returns from `session.lastBookmark()`. However, for the case if a user would somehow pass; I've added a new test case that would test nested arrays with `null` and `undefined` inside.